### ISSUE TITLE
Can't export to tiled CityGML files

### ISSUE THREAD

**Issue by nieswand** [3dcitydb/importer-exporter#295]
Hello! I'm currently trying to tile a large CityGML file into several smaller tiles (1km x 1km) using 3D City Database Importer/Exporter.
I set up the PostgreSQL database on Windows using the latest Docker file (3dcitydb/3dcitydb-pg:latest pulled today) and successfully imported my CityGML file into the database using the Importer/Exporter that I installed via the latest installer ([3DCityDB-Importer-Exporter-5.4.0-Setup.jar](https://github.com/3dcitydb/3dcitydb-suite/releases/download/v2023.0.0/3DCityDB-Importer-Exporter-5.4.0-Setup.jar)).

When trying to export the database, I get the following errors in my Docker log:
```
[335] ERROR:  table "tmp_ga89233080239" does not exist
[335] STATEMENT:  drop table TMP_GA89233080239
[335] ERROR:  current transaction is aborted, commands ignored until end of transaction block
[335] STATEMENT:  create unlogged table TMP_GA902021967086 (ID BIGINT)
```
To me, this looks like an issue with the capitalization in the name of the temporary table that is being created.
I can export the data without problems when I'm not using the tiling option.

**Comment by clausnagel** [comment]
Seems to be Docker-related. At least I have never experienced this issue with "normal" PostgreSQL installations. @BWibo, have you come across this issue before?

As a workaround in the meantime, you can let the Importer/Exporter write temporary data to a local file rather than using temporary database tables. Go to "Preferences -> General -> Cache" and pick "Use local folder". More information can be found in the [online documentation](https://3dcitydb-docs.readthedocs.io/en/version-2023.0/impexp/preferences/cache.html).

**Comment by nieswand** [comment]
Changing the setting in the preferences didn't change anything.
Neither did installing PostgreSQL directly on Windows and running the provided scripts to set up the database.
The import of the data into the database still worked fine, so I guess the database is set up correctly.

**Comment by nieswand** [comment]
Okay, there seems to be something fishy with the imported dataset. I can't reproduce the error with another dataset. My bad for not checking that prior to opening an issue. Sorry!

**Comment by clausnagel** [comment]
Thanks for your update. Would be great if you could share (part of) the dataset so that we can debug the issue and find a fix.

**Comment by nieswand** [comment]
The dataset can be downloaded [here](https://geoserver.aachen.de/legends/3dstadtmodellAC_citygml.zip). At the end of the CityGML tree, there are two elements with the tag `app:appearanceMember` which gave me errors while importing because they link to texture files that aren't present. So I removed them manually before importing the dataset. But removing them did not have any effect on the exporting issue.

I built the databse with the EPSG 25832 and the height EPSG 7837.
For the export, I used the bounding box: `bbox=287000.0,5618000.0,303000.0,5638000.0&3dcitydb_srs=UUID_2515d669-cff1-444e-a105-683f651b3fd3` and tiled it into 20 rows and 16 columns.

**Comment by clausnagel** [comment]
[like]  Claus Nagel reacted to your message:
________________________________

**Comment by clausnagel** [comment]
Thanks for providing the dataset. We can reproduce the issue and will work on a bugfix.

There is another workaround you can use. The issue is caused by the global appearances in the file. The CityGML specification recommends avoiding global appearances in favor of local appearances whenever possible. So, you can use
[citygml-tools](https://github.com/citygml4j/citygml-tools) to automatically convert the global appearances. Simply use the `to-local-apps` command of citygml-tools:

    $ citygml-tools to-local-apps /path/to/your/city.gml

Afterwards, load the resulting file into the 3DCityDB and re-export applying your tiling scheme. I just tested locally and it worked fine.

**Comment by nieswand** [comment]
That workaround works indeed, thank you!

**Comment by clausnagel** [comment]
Fixed with 4484b4f9ec3b48327ca07c480b5917bdcd0b4cea. The fix will be included in the next release. If you need the fix earlier, you can [build the Importer/Exporter](https://github.com/3dcitydb/importer-exporter?tab=readme-ov-file#building) from `master` or use the `edge` [Docker image](https://github.com/3dcitydb/importer-exporter?tab=readme-ov-file#using-with-docker).
